### PR TITLE
Implementation of "measure_dihedrals" function

### PR DESCRIPTION
**Pull request by hyejang**
I implemented 'measure_dihedrals' function in dihedral_scanner.py and add a test function to check if the function works in a correct way.

### CODE REVIEW

File: torsiondrive/dihedral_scanner.py [hunk 1]
```diff
@@ -166,13 +169,79 @@ def build_dihedral_mask(self, dihedral_ranges):
...
+
+            n1 = np.cross(v1, v2)
+            norm1 = np.linalg.norm(n1)
+            if norm1 != 0:
+                n1 /= norm1
+            n2 = np.cross(v2, v3)
+            norm2 = np.linalg.norm(n2)
+            if norm2 != 0:
+                n2 /= norm2
+            m1 = np.cross(v2, n1) / np.linalg.norm(v2)
```
Comment (dgasmith) [inline torsiondrive/dihedral_scanner.py]: Can you make sure this definition is equivalent to the QCElemental/geometric definition? I know Lee-Ping and I were opposite at one point.

**Comment by yudongqiu** [comment]
Congratulations for making the first PR! The tests can pass and the code generally looks great. 

Two things can be improved here, according to the feature I described in #49:
1. The `measure_dihedrals` function can be put outside the DihedralScanner class, as a stand-alone function. You can see that `self` is not used at all in this function.
2. After the above change is made, the unittest can be simplified--you don't need to create such DihedralScanner object to be able to test this function. It will also be nice that if you can add 5-10 more test calls to this function with different geometries.
For example, CIS/Trans conformers, and a few random geometries. Check result values using the `np.allclose` function, with `atol=1e-6`.
and test calls for these edge cases:
a) Input dihedrals not valid. -> raise Error; This could happen when dihedrals in the input list do not have 4 integers, or when the atom indices are out of bound. The ValueError should print the input dihedral_list in the error message.
b) Atoms (i-j-k) or (j-k-l) is on a straight line: trigger warning with `check_linear==True`, don't trigger warning with `check_linear!=True`; 
c) Atoms (i-j) or (j-k) or (k-l) have exactly the same position, this should lead to straight angle and trigger same warning as b)
d) Measure dihedral when not all atoms are bonded, trigger warning when `check_bonded==True`, don't trigger warning when `check_bonded!=True`.

**Comment by yudongqiu** [comment]
Also, it looks like the preferred way of calling the `warnings.warn` function is 
```
warnings.warn(message, UserWarning)
```
See here: https://docs.python.org/3/library/warnings.html#warnings.warn

File: torsiondrive/dihedral_scanner.py (hunk 1 above)
Comment (yudongqiu) [reply torsiondrive/dihedral_scanner.py]: We are planning to reproduce the values of the `Molecule.measure_dihedrals()` function so it will keep the result consistent.

**Comment by hyejang** [comment]
I separated the measure_dihedrals function from scanner class as you suggested and  added four more molecules for testing the function. I'm not sure if the test molecules are diverse enough.

**Comment by yudongqiu** [comment]
Some timings for the improvements that might be interesting:
On benzene, measuring two dihedrals:
1. original `m.measure_dihedrals()`: 237 us (with bond check, no linear angle check).
2. new `measure_dihedrals()`: 463 us with both checks, 208 us without checks.
3. optimized `measure_dihedrals()`: 32 us with both check, 14 us without checks.
